modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/CompilerWrapper.java

```diff
@@ -123,7 +123,7 @@ public boolean isOutDated() {
 
 	private URL _getClassURL(String className) {
 		String classNamePath = className.replace(
-			CharPool.PERIOD, CharPool.SLASH);
+			CharPool.PERIOD, CharPool.BACK_SLASH);
```

Comment:
@petershin This will fix the path fro Windows, but will break it for Linux. Can you change `CharPool.BACK_SLASH` to `File.separator`? That should fix it on both platforms.